include/envoy/http/async_client.h

```diff
@@ -50,18 +83,64 @@ class AsyncClient {
     virtual void cancel() PURE;
   };
 
+  /**
+   * An in-flight HTTP stream.
+   */
+  class Stream {
+  public:
+    virtual ~Stream() {}
+
+    /***
+     * Send headers to the stream.
+     * @param headers supplies the headers to send.
+     * @param end_stream supplies whether this is a header only request.
+     */
+    virtual void sendHeaders(HeaderMap& headers, bool end_stream) PURE;
+
+    /***
+     * Send data to the stream.
+     * @param data supplies the data to send.
+     * @param end_stream supplies whether this is the last data.
+     */
+    virtual void sendData(Buffer::Instance& data, bool end_stream) PURE;
+
+    /***
+     * Send trailers. This implicitly ends the stream.
+     * @param trailers supplies the trailers to send.
+     */
+    virtual void sendTrailers(HeaderMap& trailers) PURE;
+
+    /***
+     * Reset the stream.
+     */
+    virtual void reset() PURE;
```

Comment:
Does this call trigger StreamCallbacks.onResetStream or does that only occur if the RESET is received from the remote side. If we're thread local then it seems like it should call.